File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.py
from collections import Counter,defaultdict,deque;
class Solution(object):
    def smallestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        mp=Counter(s)
        left=""
        middle=""
        for ch in 'abcdefghijklmnopqrstuvwxyz':
            while(mp[ch]>=2):
                left+=ch
                mp[ch]-=2
            if(mp[ch]==1):
                middle+=ch
        res=left[::-1]
        
        return left+middle+res







        

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        int n=s.length();
        string left="";
        string middle="";
        if(n==1) return s;
        unordered_map<char,int> mp;
        for(char ch:s){
            mp[ch]++;
        }
        for(char i='a';i<='z';i++){
            while(mp[i]>=2){
                left+=i;
                mp[i]-=2;
            }
            if(mp[i]==1){
                middle+=i;
            }
        }
        string res=left;
        reverse(res.begin(),res.end());
        return left+middle+res;
    }
};